src/cls/IPM/General/LockFile.cls

```diff
@@ -0,0 +1,124 @@
+Class %IPM.General.LockFile
+{
+
+/// Name of the JSON lock file
+Parameter LockFileName As String = "module-lock.json";
+
+/// For a given module name, this will write it's lock file and save it under the module's root
+ClassMethod WriteToLockFile(
+	baseModName As %String,
+	ByRef params) As %Status
+{
+    set sc = $$$OK
+    try {
+        set verbose = $get(params("Verbose"), 0)
+        set module = ##class(%IPM.Storage.Module).NameOpen(baseModName)
+        do module.BuildDependencyGraph(.tDependencyGraph)
+
+        if (verbose) {
+            write !, "Creating lock file for module "_module.ExternalName
+        }
+
+        // Iterate over module dependency graph
+        // For each module:
+        //      1. Set values for module name, version, repository (name), and direct dependencies
+        //      2. Add repository information to the list (if the module is from a new repo)
+        set lockFileRepositories = {}
+        set lockFileDependencies = {}
+        set moduleName = ""
+        for {
+            set moduleName = $order(tDependencyGraph(moduleName))
+            quit:moduleName=""
+            set mod = ##class(%IPM.Storage.Module).NameOpen(moduleName)
+
+            if (verbose) {
+                write !, "Adding "_mod.ExternalName_" to the lock file"
+            }
+
+            // DEPENDENCY STUFF
+            set dependencyVal = {}
+            set dependencyVal.version = mod.VersionString
```

Comment:
So although we don't need persistence, it may be a good idea to create registered objects that can then be exported to JSON so there is good auto-complete and a well defined schema for the lock file.